packages/adapter-cloudflare/index.js

```diff
@@ -220,3 +234,40 @@ _headers
 _redirects
 `;
 }
+
+/**
+ * @param {string} config_file
+ * @returns {import('wrangler').Unstable_Config}
+ */
+function validate_config(config_file = undefined) {
+	const wrangler_config = unstable_readConfig({ config: config_file });
+
+	// we don't support workers sites
+	if (wrangler_config.site) {
+		throw new Error(
+			`You must remove all \`site\` keys in ${wrangler_config.configPath}. Consult https://svelte.dev/docs/kit/adapter-cloudflare#Migrating-from-Workers-Sites-to-Workers-Static-Assets`
+		);
+	}
+
+	// probably deploying to Cloudflare Pages
+	if (!wrangler_config.configPath || wrangler_config.pages_build_output_dir) {
```

Comment:
above you have the same sort of logic (https://github.com/sveltejs/kit/pull/13634/files#diff-78806028d48ad47894445421562b7796c53178948c017e13d4550def4193903eR31-R32)

Could a single shared utility be used instead?